wire-ios-request-strategy/Sources/Payloads/Payload+ConversationTests.swift

```diff
@@ -76,6 +76,72 @@ class Payload_ConversationTests: MessagingTestBase {
         XCTAssertEqual(conversation?.legacyAccessRole, nil)
     }
 
+    func test_Conversation_DecodesCipherSuite_APIVersionV4() throws {
+        // GIVEN
+        let payload = [
+            "cipherSuite": 0
+        ]
+
+        let data = try JSONSerialization.data(withJSONObject: payload, options: [])
+
+        // WHEN
+        let conversation = Payload.Conversation(data, apiVersion: .v4)
+
+        // THEN
+        XCTAssertNotNil(conversation)
+        XCTAssertNil(conversation?.cipherSuite)
+    }
+
+    func test_Conversation_DecodesCipherSuite_APIVersionV5() throws {
+        // GIVEN
+        let payload = [
+            "cipher_suite": 0
+        ]
+
+        let data = try JSONSerialization.data(withJSONObject: payload, options: [])
+
+        // WHEN
+        let conversation = Payload.Conversation(data, apiVersion: .v5)
+
+        // THEN
+        XCTAssertNotNil(conversation)
+        XCTAssertEqual(conversation?.cipherSuite, 0)
+    }
+
+    func test_Conversation_DecodesEpochTimestamp_APIVersionV4() throws {
+        // GIVEN
+        let payload = [
+            "epoch_timestamp": "2021-05-12T10:52:02.671Z"
+        ]
+
+        let data = try JSONSerialization.data(withJSONObject: payload, options: [])
+
+        // WHEN
+        let conversation = Payload.Conversation(data, apiVersion: .v4)
+
+        // THEN
+        XCTAssertNotNil(conversation)
+        XCTAssertNil(conversation?.epochTimestamp)
+    }
+
+    func test_Conversation_DecodesEpochTimestamp_APIVersionV5() throws {
+        // GIVEN
+        let payload = [
+            "epoch_timestamp": "2021-05-12T10:52:02.671Z"
+        ]
+
+        let data = try JSONSerialization.data(withJSONObject: payload, options: [])
+
+        // WHEN
+        let conversation = Payload.Conversation(data, apiVersion: .v5)
+
+        // THEN
+        XCTAssertNotNil(conversation)
+        XCTAssertEqual(conversation?.epochTimestamp, Date(timeIntervalSince1970: 1620816722.671))
```

Comment:
What was the reason for using a date like this? `Date(timeIntervalSince1970: 1620816722.671)`
Is it that date encoding / decoding isn't precise enough?

Reply:
I use the Foundation `Date` initialiser, because I want to compare the encoded string (line 130) with the value that I expect.
There are no other initialisers I know, that not use more logic like encoding etc. so there is a risk of having more bugs etc, right? e.g. if you use `conversation?.epochTimestamp?.transportString()` you don't see if encoding and decoding might have the same bug.
You can use a website like https://www.unixtimestamp.com (or the tool I like _DevUtils_) to transfer between unix timestamp and readable date.